script.js

```diff
@@ -1,12 +1,377 @@
+
+  let allResultsOnPage = Array.from(getResultsOnPage);
+  let newResults = [];
+
+  allResultsOnPage.forEach(result => {
+    if (result.innerText.toUpperCase().includes(searchString)) {
+      result.classList.remove("hidden")
+      newResults.push(result);
+    } else {
+      result.classList.add("hidden")
+    }
+
+    searchResult.innerText = `Displaying ${newResults.length}/${getResultsOnPage.length} results`;
+
+    if (searchString.length === 0) {                                                                      //I have this condition, so when you type "winter" for example and get a result of 10/73, then when you delete it all you want the result to be 73 and not 73/73
+      searchResult.innerText = `Displaying ${getResultsOnPage.length} results`;
+    }
+  });
 }
 
-window.onload = setup;
+//level 300
+
+function addEpisodesSelector() {
+  let selectAnEpisode = document.createElement("select");
+  selectAnEpisode.setAttribute("id", "episodes")
+
+  let pageNavigation = document.querySelector("nav")                                                      //I declared this variable again (there is one in level 200), did not want to have anything in the global scope, so not sure if it is best practice?
+  pageNavigation.prepend(selectAnEpisode);
+
+  let getEpisodesTitles = document.getElementsByClassName("episodeName")
+
+  let episodeOption = document.createElement("option");
+  episodeOption.innerText = "All Results";
+  selectAnEpisode.appendChild(episodeOption);
+
+  for (let titleIndex = 0; titleIndex < getEpisodesTitles.length; titleIndex++) {
+    episodeOption = document.createElement("option");
+    selectAnEpisode.appendChild(episodeOption);
+
+    episodeOption.innerText = getEpisodesTitles[titleIndex].innerText;
+  }
+
+  selectAnEpisode.addEventListener("change", selectedEpisode)
+}
+
+function selectedEpisode(e) {
+  let selectedOption = e.target.value;
+  let getResultsOnPage = document.getElementsByClassName("wrapper");                             //I declared this variable again (there is one in level 200), did not want to have anything in the global scope, so not sure if it is best practice?
+  let allResultsOnPage = Array.from(getResultsOnPage);
+
+  allResultsOnPage.forEach(result => {
+    if (result.innerText.includes(selectedOption)) {
+      result.classList.remove("hidden");
+    } else {
+      result.classList.add("hidden");
+    }
+
+    if (selectedOption === "All Results") {
+      result.classList.remove("hidden");
+    }
+
+  })
+}
+
+//Level 400
+
+function addShowsSelector() {
+  let selectAShow = document.createElement("select");
+  selectAShow.setAttribute("id", "shows");
+
+  let chooseEpisodeAndShow = document.createElement("label");
+  chooseEpisodeAndShow.innerText = "Choose a show";
+  chooseEpisodeAndShow.classList.add("chooseLabel")
+
+  let pageNavigation = document.querySelector("nav")
+  pageNavigation.prepend(chooseEpisodeAndShow, selectAShow);
+
```

Comment:
Lines 207 - 213 are the same as 245 - 251. You can avoid this duplicated code by just calling the `sortedShowsList` function here.